apps/src/lib/levelbuilder/data-docs-editor/DataDocEditAll.jsx

```diff
@@ -1,11 +1,103 @@
-import React from 'react';
+import React, {useState} from 'react';
+import $ from 'jquery';
+import PropTypes from 'prop-types';
+import {Link, TextLink} from '@dsco_/link';
+import FontAwesome from '@cdo/apps/templates/FontAwesome';
+import Dialog, {Title as DialogTitle} from '@cdo/apps/templates/Dialog';
 
 const DataDocEditAll = props => {
+  const {dataDocs: initialDataDocs} = props;
+  const [dataDocs, setDataDocs] = useState(initialDataDocs);
+  const [showDeleteWarningDialog, setShowDeleteWarningDialog] = useState(false);
+  const [pendingDeleteDocKey, setPendingDeleteDocKey] = useState(null);
+
+  const initiateDeleteDataDoc = docKeyToDelete => {
+    setPendingDeleteDocKey(docKeyToDelete);
+    setShowDeleteWarningDialog(true);
+  };
+
+  const deleteDataDoc = () => {
+    $.ajax({
+      url: `/data_docs/${pendingDeleteDocKey}`,
+      method: 'DELETE',
+      success: () => deleteRequestCleanup(),
+      error: (xhr, ajaxOptions, thrownError) => {
+        if (xhr.status === 404) {
+          deleteRequestCleanup();
+        }
+      }
+    });
+  };
+
+  const deleteRequestCleanup = () => {
+    setDataDocs([
+      ...dataDocs.filter(dataDoc => !pendingDeleteDocKey.includes(dataDoc.key))
+    ]);
+    setShowDeleteWarningDialog(false);
+    setPendingDeleteDocKey(null);
+  };
+
   return (
     <div>
-      <h1 style={{marginBottom: 30}}>Data Docs Edit All Page</h1>
+      <h1 style={{marginBottom: 30}}>Data Docs</h1>
+      <div className="page-actions">
+        <TextLink
+          className="create-btn"
+          icon={<FontAwesome icon="plus" />}
+          iconBefore={true}
+          href={'/data_docs/new'}
+          text="Create New Data Doc"
+        />
+      </div>
+      {showDeleteWarningDialog && (
+        <Dialog
+          body={
+            <DialogTitle>{`Are you sure you want to permanently delete data doc ${pendingDeleteDocKey}?`}</DialogTitle>
+          }
+          cancelText="Cancel"
+          confirmText="Delete"
+          confirmType="danger"
+          isOpen={true}
+          handleClose={() => setShowDeleteWarningDialog(false)}
+          onCancel={() => setShowDeleteWarningDialog(false)}
+          onConfirm={() => deleteDataDoc()}
```

Comment:
one late tip here - this approach is A+, but also a simple `window.confirm` is sufficient for confirming levelbuilder actions in the future.

Reply:
Gotcha, I'll keep that in mind for the future thank you!